### PR TITLE
#5934 changing system.outs to log messages 

### LINKED ISSUE #5934: [BUG] Change the Subject Area FVT to use logger rather than System.err / System.out
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

A large amount of output goes to the log, this should not happen by default.

### Expected Behavior

much less content in the output of the FVT and the build bu default

Switch on debug in the logging configuration and then see all the output.

### Steps To Reproduce

run the build

### Environment

```markdown
- Egeria:
- OS:
- Java:
- Browser (for UI issues):
- Additional connectors and integration:
```

### Any Further Information?

This should be reasonably easy to change for a Java developer. By looking in the Subject Area FVT project and defining a logger per class and replacing  System.err and System.out to logger.debug or logger.error as appropriate. 

an example of the usage of stdout is 
[https://github.com/odpi/egeria/blob/master/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/GraphFVT.java](https://github.com/odpi/egeria/blob/master/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/GraphFVT.java)

An example of what should be changed is 
`System.out.println("Create a glossary");`

There is code which produces output to stdout used when prompting for user input. This code should not be changed. 
for example [https://github.com/odpi/egeria/blob/master/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/RunAllFVT.java](https://github.com/odpi/egeria/blob/master/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/RunAllFVT.java)

`public static String getServerName(String[] args) throws IOException
    {
        String name = null;
        if (args.length > 1)
        {
            name = args[1];
        } else
        {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            System.out.print("Enter a OMAG Server Name. Press enter to get the default (" + FVTConstants.SERVER_NAME1 + ")) :");
            name = br.readLine();
            if (name.equals(""))
            {
                name = FVTConstants.SERVER_NAME1;
            }

        }
        return name;
    }`

Also all of the main methods System.out should not be removed.

### PR DESCRIPTION
**Pull request by odttlnt**
- Updated to use log framework
- updated dependencies for Log framework

Signed-off-by: Sachin Naik <91499691+odttlnt@users.noreply.github.com>

# Description

The subject author FVT has been updated to use the log framework rather than System. 
System. has been kept only on the Main functions and at messages indicating start and end of individual FVTs

Fixes #5934 

# How Has This Been Tested?
mvn clean install , gradle build and Intellij runs. Plus some runs to check log levels.

# Any additional notes for reviewers?
ForCast compile error is expected in maven compile

### CODE REVIEW

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/ConfigFVT.java [hunk 1]
```diff
@@ -35,12 +39,12 @@ public static void main(String args[])
         } catch (SubjectAreaFVTCheckedException e) {
             System.out.println("ERROR: " + e.getMessage() );
         } catch (UserNotAuthorizedException | InvalidParameterException | PropertyServerException e) {
-            System.out.println("ERROR: " + e.getReportedErrorMessage() + " Suggested action: " + e.getReportedUserAction());
+            log.debug("ERROR: " + e.getReportedErrorMessage() + " Suggested action: " + e.getReportedUserAction());
```
Comment (davidradl) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/ConfigFVT.java:42]: I suggest log.error for all errors using one of the overloaded methods that include the Throwable

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml [hunk 2]
```diff
@@ -50,8 +50,24 @@
         <dependency>
             <groupId>org.odpi.egeria</groupId>
             <artifactId>subject-area-client</artifactId>
+<!--
```
Comment (davidradl) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml:53]: why have we added this comment?

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml [hunk 3]
```diff
@@ -50,8 +50,24 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <scope>compile</scope>
         </dependency>
 
+
+<!--        <dependency>
```
Comment (davidradl) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml:65]: why have we left this deletion in?

**Review by davidradl — CHANGES_REQUESTED** [review]

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/CategoryFVT.java [hunk 4]
```diff
@@ -95,12 +98,12 @@ public static int getCategoryCount(String url, String serverName, String userId)
 
     public void run() throws  SubjectAreaFVTCheckedException, InvalidParameterException, PropertyServerException, UserNotAuthorizedException {
 
-        System.out.println("Create a glossary");
+        log.debug("Create a glossary");
         Glossary glossary = glossaryFVT.createGlossary(serverName + " " + DEFAULT_TEST_GLOSSARY_NAME);
         String glossaryGuid = glossary.getSystemAttributes().getGUID();
-        System.out.println("Create a category1");
+        log.debug("Create a category1");
```
Comment (davidradl) [inline open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/CategoryFVT.java:104]: can we surround all log.debug with if (log.isDebugEnabled()) so the parameters are no formatted when debug is not on please.

**Review by davidradl — CHANGES_REQUESTED** [review]

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml (hunk 3 above)
Comment (odttlnt) [reply open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/pom.xml:65]: Deleted

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/CategoryFVT.java (hunk 4 above)
Comment (odttlnt) [reply open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/CategoryFVT.java:104]: Completed

File: open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/ConfigFVT.java (hunk 1 above)
Comment (odttlnt) [reply open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/fvt/ConfigFVT.java]: This has been taken care of. Except for the throws for IO Exception

**Pull request merged**